### ISSUE TITLE
error in installing nginx-1.12.2 on centos 7

### ISSUE THREAD

**Issue by carolfly86** [sous-chefs/nginx#441]
### Cookbook version
7.0.0

### Chef-client version
[Version of chef-client in your environment]

### Platform Details
Centos 7

### Scenario:
installing nginx in package method

### Steps to Reproduce:
include_recipe 'nginx'

### Expected Result:
nginx package should be installed
Although the default node['nginx']['version'] = 1.12.1, it still tries to install 1.12.2.

### Actual Result:

```
 [2017-10-18T14:31:40+00:00] ERROR: yum_package[nginx] (nginx::package line 43) had an error: Chef::Exceptions::Exec: yum -d0 -e0 -y --disablerepo=* --enablerepo=nginx install nginx-1.12.2-1.el7_4.ngx returned 1:
       STDOUT:  You could try using --skip-broken to work around the problem
        You could try running: rpm -Va --nofiles --nodigest
       STDERR: Error: Package: 1:nginx-1.12.2-1.el7_4.ngx.x86_64 (nginx)
           Requires: libcrypto.so.10(OPENSSL_1.0.2)(64bit)
```

**Comment by josh-barker** [comment]
Hi @carolfly86 

I've found the same problem.

In the mean time, you can use something like this:

```ruby
edit_resource(:package, 'nginx') do
  version '1.12.1-1.el7.ngx'
end
```

**Comment by jalkoby** [comment]
I fixed this issue by running 'yum update' before installing nginx

**Comment by fletchowns** [comment]
This one sounds similar to a ticket I filed, #446

In my case, the nginx cookbook installed 1.12.1 and then I did a `yum update` which installed 1.12.2. Now, the nginx cookbook is still trying to install 1.12.1 which fails "installed package is newer than candidate package". The weird part is that it fails even if I specify `default['nginx']['version'] = '1.12.2'`.

**Comment by fletchowns** [comment]
Nevermind... #446 was bogus, forgot I had put in the `edit_resource` workaround from this ticket. That's why it was still trying to install the older version 😊

**Comment by Iakim** [comment]
I fixed this issue by running yum update before installing nginx ² 628 packages... hahahaha, I needed to change without satellite library to an environment. 😊

**Comment by jahrensnv** [comment]
I got around this by updating just the `nss` and `openssl` packages first.

**Comment by fxrsliberty** [comment]
if you have the cookbook openssl and add the openssl::upgrade cookbook ahead of the nginx cookbook in a role then it works on RHEL / CentOS 7.3

**Comment by damacus** [comment]
Closing due to inactivity. 

If this is still an issue please reopen or open another issue. Alternatively drop by the #sous-chefs channel on the [Chef Community Slack](http://community-slack.chef.io/) and we'll be happy to help! 

Thanks,
Sous-Chefs
